### PR TITLE
Character Pages: Talents and Constellation Tabs

### PR DESCRIPTION
**Pull request by riceconfetti**

### CODE REVIEW

File: src/lib/components/ui/builds/artifact-item.svelte [hunk 1]
```diff
@@ -15,11 +15,11 @@
 {#if signature}
 	<div class="flex gap-2 w-full">
 		<img
-			class="flex-grow aspect-square object-contain h-0 min-h-full"
+			class="basis-1/4 aspect-square object-contain w-full"
```
Comment (maalni) [inline src/lib/components/ui/builds/artifact-item.svelte:18]: Would add size-14 since image is pretty huge.

Screenshot 2024-04-02 215836

Comment (maalni) [reply src/lib/components/ui/builds/artifact-item.svelte:18]: Text is also still clipping on smaller screens

File: src/routes/characters/[slug]/+page.svelte [hunk 2]
```diff
@@ -19,111 +20,67 @@
...
-	const splash = S3Service.getCharacterLink(data.character) + '/gacha_splash.webp';
-	const ascensions = data.characterData.ascension;
+	const card =
+		S3Service.getCharacterLink(data.character) +
+		'/gacha_card.' +
+		data.characterData.pictures.gatchaCard.split('.')[1];
+	const splash =
+		S3Service.getCharacterLink(data.character) +
+		'/gacha_splash.' +
+		data.characterData.pictures.gachaSplash.split('.')[1];
```
Comment (maalni) [inline src/routes/characters/[slug]/+page.svelte:30]: only .webp

File: src/routes/characters/[slug]/temp
Comment (maalni) [inline src/routes/characters/[slug]/temp:1]: Should this be commited?

File: src/lib/components/ui/character-tabs/talent-table.svelte
Comment (maalni) [inline src/lib/components/ui/character-tabs/talent-table.svelte:1]: Please move this to components/table

File: src/lib/components/ui/character-tabs/talent-table.svelte [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			Skill Attributes: (Level {talentLevels[currentSkill] + 1})
+		</Text>
+		<Slider
+			value={[10]}
+			onValueChange={(v) => {
+				talentLevels[currentSkill] = v[0];
+			}}
+			max={14}
+			step={1}
+		/>
```
Comment (maalni) [inline src/lib/components/ui/character-tabs/talent-table.svelte:107]: Would move this underneath the table and restrict its length to the table section

File: src/lib/components/ui/character-tabs/talent-table.svelte [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	</div>
+	<div class="flex flex-1 flex-col gap-2 lg:flex-row lg:max-h-full">
+		{#each passives as passive}
+			<Card class="flex flex-col bg-neutral border-secondary w-full lg:overflow-auto">
+				<Collapsible.Root>
+					<Collapsible.Trigger>
+						<Text type="p" class="font-bold text-lg">
+							{`${passive.level == 0 ? 'Utility Passive: ' : `A${passive.level}: `} ${passive.name}`}
+						</Text>
+					</Collapsible.Trigger>
```
Comment (maalni) [inline src/lib/components/ui/character-tabs/talent-table.svelte:48]: The trigger should grow. Otherwise opening the collapsible is unreliable for some passives

**Review by maalni — CHANGES_REQUESTED** [review]

File: src/routes/characters/[slug]/temp
Comment (riceconfetti) [reply src/routes/characters/[slug]/temp:1]: oh whoops, no I'll delete that

File: src/lib/components/ui/builds/artifact-item.svelte (hunk 1 above)
Comment (riceconfetti) [reply src/lib/components/ui/builds/artifact-item.svelte:18]: What screen size is the text clipping on? I can't seem to get it them to clip when I resize the page.

File: src/lib/components/ui/character-tabs/talent-table.svelte (hunk 4 above)
Comment (riceconfetti) [reply src/lib/components/ui/character-tabs/talent-table.svelte:48]: What do you mean by this? Would changing flex-row to flex-col be better for the ui overall or should I still try to keep the passives in a single row?

Comment (maalni) [reply src/lib/components/ui/character-tabs/talent-table.svelte:48]: Screenshot 2024-04-06 152238

The collapsible is only triggered when clicking on the text area. This makes it hard to open when it doesnt fill the whole space

File: src/lib/components/ui/builds/artifact-item.svelte (hunk 1 above)
Comment (maalni) [reply src/lib/components/ui/builds/artifact-item.svelte:18]: Screenshot 2024-04-06 152709

Clips on widths larger than 1024px (until screen is large enough)

Comment (maalni) [reply src/lib/components/ui/builds/artifact-item.svelte:18]: Im using an 21:9 display if this helps

Comment (riceconfetti) [reply src/lib/components/ui/builds/artifact-item.svelte:18]: Ah I see, it doesn't happen on Firefox. Looks like adding size-14 fixes it but I'll test the rest of the page on Chrome to check that the overflow doesn't happen on anything else.

**Review by maalni — APPROVED** [review]
